File: Daily Test/Python/Smallest Binary Number Sum 26.12.2025.py
n = int(input())
sum_ = 0

for _ in range(n):
    x = int(input())
    k = 1
    while True:
        bin_str = bin(k)[2:]
        dec = int(bin_str)
        if dec % x == 0:
            sum_ += k
            break
        k += 1

print(sum_)

File: Daily Test/Java/Smallest Binary Number Sum 26.12.2025.java
import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int n = sc.nextInt();
        int sum = 0;

        for (int i = 0; i < n; i++) {
            int x = sc.nextInt();
            int k = 1;

            while (true) {
                String bin = Integer.toBinaryString(k);
                long dec = Long.parseLong(bin);

                if (dec % x == 0) {
                    sum += k;
                    break;
                }
                k++;
            }
        }

        System.out.println(sum);
    }
}